src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -681,58 +701,61 @@ impl BinaryValueWriter_1_1<'_, '_> {
         const MIN_20_BIT_ADDRESS: usize = 4_160;
         const MAX_20_BIT_ADDRESS: usize = 1_052_735;
 
-        // Try to get an address from the provided MacroIdLike-thing.
-        let id_ref: MacroIdRef<'_> = macro_id.prefer_address();
-        match id_ref {
-            // We can't invoke e-expressions by name in binary, so if it can't provide an address that's a problem.
-            MacroIdRef::LocalName(_name) => {
-                return IonResult::encoding_error(
-                    "the raw binary writer cannot encode macros by name",
-                );
-            }
-            MacroIdRef::LocalAddress(address) if address <= MAX_4_BIT_ADDRESS => {
-                // Invoke this ID with a one-byte opcode
-                self.encoding_buffer.push(address as u8);
-            }
-            MacroIdRef::LocalAddress(address) if address <= MAX_12_BIT_ADDRESS => {
-                // Encode as an opcode with trailing 1-byte FixedUInt
-                const BIAS: usize = MIN_12_BIT_ADDRESS;
-                let biased = address - BIAS;
-                let low_opcode_nibble = biased >> 8;
-                self.encoding_buffer.extend_from_slice_copy(&[
-                    // Opcode with low nibble setting bias
-                    0x40 | low_opcode_nibble as u8,
-                    // Remaining byte of magnitude in biased address
-                    biased as u8,
-                ]);
+        let macro_ref = macro_id.resolve(self.macros)?;
+        let address = macro_ref.address();
```

Comment:
🪧 Now that the raw writers have access to the macro table, they can do their own ID resolution. This means that the binary writer no longer has to handle the case where the caller provided a text macro ID that may or may not be valid. The ID can either be mapped to an address or it's an error.